### PR TITLE
Create the push-credentials Secret on behalf of the user

### LINKED ISSUE #718: Create the push-credentials secret for me
While `riff namespace init` goes a long way towards making setup up a new namespace easier, it's still painful to correctly setup push credentials for your registry of choice and very error prone.

As an alternative to specifying a custom `--secret`, we should create a secret for the user from credentials they provide.

For GCR:
```bash
$ riff namespace init default --gcr ./path/to/credentials.json
```

For Docker Hub prompt for the password:
```bash
$ riff namespace init default --dockerhub username
Docker Hub password: *******
```

For non-interactive Docker Hub use accept password via stdin:
```bash
$ pbpaste | riff namespace init default --dockerhub username
```

Any user that isn't comfortable giving the riff cli their credentials can still create the secret manually. We can extend support to other registry types in the future.

### LINKED ISSUE #721: riff init namespace doesn't apply files in the correct namespace

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes #718
Fixes #721

### CODE REVIEW

File: pkg/core/namespace.go [hunk 1]
```diff
@@ -48,47 +64,109 @@ func (kc *kubectlClient) NamespaceInit(options NamespaceInitOptions) error {
...
 		fmt.Printf("%s\n", log)
 		if err != nil {
 			return err
 		}
 	}
-
 	return nil
 }
+
+func randomName() string {
```
Comment (trisberg) [inline pkg/core/namespace.go]: doesn't seem very random, always generates "push-credentials-fpllngz" for me

Comment (ericbottard) [reply pkg/core/namespace.go]: Ah, go random has a fixed seed by default. Need to use timestamp

Comment (scothis) [reply pkg/core/namespace.go]: use generateName https://stackoverflow.com/a/48023670

File: pkg/core/namespace.go [hunk 2]
```diff
@@ -17,7 +17,20 @@
...
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"syscall"
+
+	"golang.org/x/crypto/ssh/terminal"
+
+	v12 "k8s.io/api/core/v1"
```
Comment (scothis) [inline pkg/core/namespace.go]: `corev1` or something like that is a more idiomatic name

**Pull request merged**